src/discord/events/message/supportRequest.ts

```diff
@@ -1,32 +1,78 @@
 import Discord from 'discord.js';
-import { SupportRequest, SupportRequestType, SupportRequestErrors } from '../../../entities/supportRequest';
+import { Not } from 'typeorm';
+import { SupportRequest, SupportRequestType, SupportRequestStatus } from '../../../entities/supportRequest';
 import logger from '../../../logger';
 import { Config } from '../../../entities/config';
+import { SubCommands } from '.';
+import { DiscordContext } from '../../../entities/discordContext';
 
-export async function supportRequest(msg: Discord.Message): Promise<void> {
-  const discordId = msg.author.id;
-  const discordName = msg.author.username;
-  const actionId = msg.content === '!technicalSupport' ? SupportRequestType.TechnicalSupport : SupportRequestType.IdeaPitch;
+/* eslint-disable no-param-reassign */
 
+interface UserInfo {
+  [k: string]: number | string | string[];
+  id: string;
+  requestType: SupportRequestType;
+  username: string;
+}
+
+enum steps {
+  inputName = 'inputName',
+}
+
+const requestTypeMapping: { [id: string]: SupportRequestType } = {
+  '!technicalSupport': SupportRequestType.TechnicalSupport,
+  '!ideaPitch': SupportRequestType.IdeaPitch,
+  '!jobChat': SupportRequestType.JobChat,
+};
+
+export async function supportRequest(msg: Discord.Message, context: DiscordContext): Promise<void> {
   const supportRequestQueueActive = await Config.findToggleForKey('supportRequestQueueActive');
   if (!supportRequestQueueActive) {
     msg.author.send("**Whoops...**\n:see_no_evil: Our team isn't available to help at the moment, check back with us soon!");
-  } else {
+    return;
+  }
+
+  const userHasOpenRequests =
+    (await SupportRequest.createQueryBuilder()
+      .where({ slackId: msg.author.id, status: Not(SupportRequestStatus.Complete) })
+      .getCount()) > 0;
+  if (userHasOpenRequests) {
```

Comment:
Thoughts on refactoring this to not have an inline `await` statement? The `> 0` is hard to see IMO. (The suggested code will probably not pass linting, so it might need to be formatted)

```suggestion
  const userOpenRequestsCount =
    await SupportRequest.createQueryBuilder()
      .where({ slackId: msg.author.id, status: Not(SupportRequestStatus.Complete) })
      .getCount();
  if (userOpenRequestsCount > 0) {
```